library/core/src/cmp.rs

```diff
@@ -851,11 +853,12 @@ pub trait Ord: Eq + PartialOrd<Self> {
     #[inline]
     #[must_use]
     #[rustc_diagnostic_item = "cmp_ord_max"]
-    fn max(self, other: Self) -> Self
+    fn max(self, other: Rhs) -> Self
     where
         Self: Sized,
+        Rhs: Sized + Into<Self>,
```

Comment:
I wasn't entirely sure what the best way to handle these methods was, but since this is guaranteed by `impl From<T> for T` in the stable case, I figured it was okay.